compiler/noirc_evaluator/src/acir/arrays.rs

```diff
@@ -241,6 +243,13 @@ impl Context<'_> {
             // as if the predicate were true. This is as if the predicate were to resolve to false then
             // the result should not affect the rest of circuit execution.
             let value = array[index].clone();
+            // An `Array` might contain a `DynamicArray`, however if we define the result this way,
+            // we would bypass the array initialization and the handling of dynamic values that
+            // happens in `array_get_value`. Rather than repeat it here, let the non-special-case
+            // handling take over by returning `false`.
+            if matches!(value, AcirValue::DynamicArray(_)) {
+                return Ok(false);
```

Comment:
Why not just initialize it here?

Reply:
I felt the opposite: by returning a flag it felt like it was looking for a quick win without trying too hard. It flat out returns false for top level DynamicArray for example. I thought if where arrays can be initialised it adds to the scattering of logic.

That said I would be in support for revisiting this part of the code because it’s a bit sparsely documented.

Reply:
> I thought if where arrays can be initialised it adds to the scattering of logic.

Yes, that is a fair point. We can leave as is for now. This code could use revisiting in general to streamline the logic.